### PR TITLE
Editing and Deleting Categories +++ Creating Reactions

### PR DESCRIPTION
**Pull request by alambrow**
## Files Modified
`CategoryList.js` and `CategoryProvider.js` +++ `ReactionBox.js` and `ReactionProvider.js`, plus `NavBar.js` and `ApplicationViews.js`

## Files Created
`/reaction/ReactionBox.js` and `reaction/ReactionProvider.js`

## Changes Made
Fixed functionality of editing and deleting categories;
Added "Reaction Manager" to view and add potential reactions to posts

To test: 
1. Clone repo (al-categories) and check that, once logged in, you can DELETE and EDIT categories (pending server-side PR from 'al-testbranch' [branch now merge into main in server repo])
2. Go to "Reaction Manager" and test that you can POST a new reaction (sample url: https://upload.wikimedia.org/wikipedia/commons/thumb/2/23/Facebook_Like_button.svg/1024px-Facebook_Like_button.svg.png)

> Thank you again to Val for helping me solve the issue with the constant re-rendering of the modal box upon text input

### CODE REVIEW

**Review by mecrowley — COMMENTED** [review]
There is a glitch with the modal input box that appears when editing the category.  Every time there is a keypress, the state seems to reset and typing cannot resume until the input box is clicked again.  Is there a way to possibly fix this?

**Comment by alambrow** [comment]
> There is a glitch with the modal input box that appears when editing the category. Every time there is a keypress, the state seems to reset and typing cannot resume until the input box is clicked again. Is there a way to possibly fix this?

Resolved! The issue lay with re-rendering. The page would re-render upon text input, causing annoying flashing. The modal component with the text input had to be invoke through a function call rather than as a React component.

**Review by enbeec — APPROVED** [review]
Looks awesome!

**Review by enbeec — APPROVED** [review]
I tested the code before I pushed it :)

File: src/components/reaction/ReactionBox.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                    ))
+                }
+                <button className="reaction_create_new" onClick={() => {setModalShow(true)}}>
+                    <BsPlusCircleFill />
+                </button>
+                {ReactionModal({
+                    show: modalShow,
+                    onHide: () => setModalShow(false),
+                })}
+}
```
Comment (enbeec) [inline src/components/reaction/ReactionBox.js:104]: found the culprit!

**Review by enbeec — APPROVED** [review]

**Review by mecrowley — APPROVED** [review]
works perfectly

**Review by enbeec — APPROVED** [review]
Thought it'd be fun to approve the removal of one (1) curly bracket.

**Pull request merged**